app/src/main/java/com/guardian/podxdemo/presentation/common/HideKeyboard.kt

```
@@ -0,0 +1,11 @@
+package com.guardian.podxdemo.presentation.common
+
+import android.app.Activity
+import android.view.inputmethod.InputMethodManager
+import androidx.fragment.app.Fragment
+
+fun Fragment.hideKeyboard() {
+    (activity?.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager).apply {
```

If we're not too careful in Kotlin we can end up with a lot of elvis operators, `?.` when we don't actually need them. 

In this example we can use `requireActivity()` rather than the references to the optional `activity` field. If this function is called and the Fragment isn't attached to an Activity, it will throw an exception, rather than failing silently. This might be preferred as it will alert you to undesired behaviour. Fewer elivs operators will make the code more readable and less error prone.

---

Yeah i definitely have had a tendency to just chuck elvis operators all over the place. Thanks for alerting me to `requireActivity()`, i'll also do a quick look over all my uses of `getActivitiy()` and `getView()` and suss out where it makes sense to use the `require` methods